app/models/post.rb

```diff
@@ -1,5 +1,17 @@
 class Post < ApplicationRecord
-  belongs_to :author, class_name: 'User'
-  has_many :comments, dependent: :destroy
-  has_many :likes, as: :likeable, dependent: :destroy
+  belongs_to :author, class_name: 'User', foreign_key: 'user_id'
+  has_many :comments
+  has_many :likes
+
+  after_create :update_user_posts_counter
```

Comment:
### [OPTIONAL]
Dear @harshi0102 
You can also use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - after_save